ops/model.py

```diff
@@ -1045,6 +1053,45 @@ def pebble(self) -> 'pebble.Client':
         """The low-level :class:`ops.pebble.Client` instance for this container."""
         return self._pebble
 
+    @contextmanager
+    def is_ready(self) -> None:
+        """Check whether or not Pebble is ready as a simple property.
+
+        :meth:`is_ready` implements a `contextmanager` which can be used in charms
+        to wrap :class:`Container` operations which depend on the Pebble backend
+        being available. When `is_ready` is used, exceptions from the underlying
+        Pebble operations will log error messages rather than raising exceptions.
+
+        Example:
+            container = self.unit.get_container("example")
+
+            with container.is_ready():
+                container.pull('/does/not/exist')
+                # This will result in an `ERROR` log from PathError, but not a backtrace
```

Comment:
Worth showing what a charm author would need to do in the case where they want to do something different if this fails? Does this have a different return code? It's not clear from this example if `is_ready()` has succeeded and `container.pull` is failing or `is_ready` has failed.